File: Iteration-2/src/Advisor.java
public class Advisor {

    private String name;

    public Advisor(String name) {
        this.name = name;

    }
    /**
     * This method checks student's credit sufficiency, prerequisite of the course that the students wants to
     * take and the collision status. It calls the methods defined below to
     * perform the checks
     * @param  student  The student who completed the course selection and sent it to the advisor for approval
     * @param  courseSection The course section which the student applies for registration
     *
     */
    public void approveCourse(Student student, CourseSection courseSection) {
        if (student.getTranscript().calculateCredits() < courseSection.getCourse().getRequiredCredits()) { // Checks credits for only engineering projects
            student.setLogString("\nInsufficient credit");
            courseSection.setNumberOfCreditFail(courseSection.getNumberOfCreditFail()+1);
            courseSection.setFailedCredit(courseSection.getFailedCredit() + " " + student.getStudentID());

        }else if (courseSection.getCourse().getPrerequisite() != null && !student.isPassed(courseSection.getCourse().getPrerequisite())) {
            student.setLogString("\nNot approved " + courseSection.getCourseSectionCode() +
                    " because student failed prerequisite: " + courseSection.getCourse().getPrerequisite().getCourseCode());
            courseSection.setNumberOfPrereqFail(courseSection.getNumberOfPrereqFail()+1);
            courseSection.setFailedPrereq(courseSection.getFailedPrereq() + " " + student.getStudentID());

        }else if (student.getSchedule().isCollision(courseSection)) {
            student.setLogString("\nNot approved. Collision!");
            courseSection.setNumberOfCollisionFail(courseSection.getNumberOfCollisionFail()+1);
            courseSection.setFailedCollision(courseSection.getFailedCollision() + " " + student.getStudentID());

        }/*else if(courseSection.getNumberOfQuotaFail() > 0){
            student.setLogString("\nThe student couldn't register for " + courseSection.getCourseSectionCode() + " because " +
                    "of a quota problem.");
            courseSection.setNumberOfQuotaFail(courseSection.getNumberOfQuotaFail()+1);
            courseSection.setFailedQuota(courseSection.getNumberOfQuotaFail() + " " + student.getStudentID());
        }*/
        else {
            courseSection.addStudent(student);
        }
        //showStatistics();
    }

    public String getName () {
        return name;
    }


}

File: Python Iteration/src/Advisor.py
import string


class Advisor(object):
    def __init__(self, name):
        self.__name = name

    def approve_course(self, student, course_section):
        if student.transcript.calculate_credits() < course_section.course.required_credits:
            student.set_log("\nInsufficient credit")
            course_section.num_of_credit_fail += 1
            #course_section.failed_credit(course_section.failed_credit + " " + student.student_id)
        elif course_section.course.prerequisite and student.is_passed(
                course_section.course.prerequisite) is False:
            student.set_log("\nNot approved " + course_section.course_section_code + " because student failed " \
                                                                                 "prerequisite: " + \
                        course_section.course.prerequisite.course_code)
            course_section.num_of_prereq_fail += 1
            #course_section.failed_prereq += course_section.failed_prereq + " " + str(student.student_id)
        elif student.schedule.is_collision(course_section):
            #student.set_log("\nNot approved. Collision!")
            course_section.num_of_collision_fail += 1
            #course_section.failed_collision += course_section.failed_collision + " " + str(student.student_id)
        elif course_section.num_of_quota_fail > 0:
            student.set_log("\nThe student couldn't register for " + course_section.course_section_code + " because of a "
                                                                                                      "quota problem.")
            course_section.num_of_quota_fail += 1
            #course_section.failed_quota += course_section.failed_quota + " " + str(student.student_id)
        else:
            course_section.add_student(student)

    @property
    def name(self):
        return self.__name
